neuralprophet/forecaster.py

```diff
@@ -1772,6 +1796,44 @@ def plot_components(
                     "df_name parameter is required for multiple time series and local modeling of at least one component."
                 )
 
+        # Check if list is passed and ensure lower case of string
+        if components is not None:
+            if type(components) is not list:
+                components = [components]
+            components = [comp.lower() for comp in components]
+
+            # Error if selected plotting panels are not configured in model or mis-spelled
+            for comp in components:
+                if (
+                    ("seasonality" in comp and self.model.config_season is None)
+                    or ("autoregression" in comp and not self.model.n_lags > 0)
+                    or ("lagged_regressors" in comp and self.model.config_lagged_regressors is None)
+                    or (
+                        "events" in comp
+                        and ("events_additive" not in fcst.columns and "events_multiplicative" not in fcst.columns)
+                    )
+                    or (
+                        "future_regressors" in comp
+                        and (
+                            "future_regressors_additive" not in fcst.columns
+                            and "future_regressors_multiplicative" not in fcst.columns
+                        )
+                    )
+                    or ("uncertainty" in comp and not len(self.model.quantiles) > 1)
```

Comment:
Idea: As the checks whether a model component is actually enabled is implemented several times and might be useful for other functionalities (e.g. configuration management) of our framework, it might be a good idea to extract the implementation and unify those checks (I stumbled across this while rewriting parts of the configurations)

Reply:
@noxan thanks for the input! I think this is a great idea. Let´s sync on this.